query/src/main/scala/filodb/query/exec/RangeVectorTransformer.scala

```diff
@@ -111,36 +138,44 @@ final case class InstantVectorFunctionMapper(function: InstantFunctionId,
 
 private class DoubleInstantFuncIterator(rows: Iterator[RowReader],
                                         instantFunction: DoubleInstantFunction,
+                                        scalar: Seq[ScalarRangeVector],
                                         result: TransientRow = new TransientRow()) extends Iterator[RowReader] {
   final def hasNext: Boolean = rows.hasNext
   final def next(): RowReader = {
     val next = rows.next()
-    val newValue = instantFunction(next.getDouble(1))
-    result.setValues(next.getLong(0), newValue)
+    val nextVal = next.getDouble(1)
+    val timestamp = next.getLong(0)
+    val newValue = instantFunction(nextVal, scalar.map(_.getValue(timestamp)))
+    result.setValues(timestamp, newValue)
     result
   }
 }
 
 private class H2DoubleInstantFuncIterator(rows: Iterator[RowReader],
                                           instantFunction: HistToDoubleIFunction,
+                                          scalar: Seq[ScalarRangeVector],
```

Comment:
Will we ever have more than one ScalarRangeVector when applying instant function ?
Cant we simplify the args ? This would apply to all the InstantFuncIterators

Reply:
never mind. It is a generic method that could apply for funcs with more than one arg. Some doc would have helped! It is very easy to forget!